### PR TITLE
{equicord, equibop}: init

### LINKED ISSUE #333175: Package request: equibop
**Project description**

A custom Discord App (a fork of [Vesktop](https://github.com/Vencord/Vesktop)) aiming to provide better performance and improve linux support. [Equicord](https://github.com/Combatmaster331/Equicord) (an enhanced version of [Vencord](https://github.com/Vendicated/Vencord)) is preinstalled. 

**Metadata**

* homepage URL: https://github.com/Equicord/equibop
* source URL: https://github.com/Equicord/equibop
* license: gpl3
* platforms: linux

### PR DESCRIPTION
**Pull request by NotAShelf**
- **equicord: init at 1.10.4**
- **equibop: init at 2.0.9**

Adds Equicord and Equibop derivations based mostly on existing work done on
Vencord and Vesktop respectively. The build process is almost exactly the same
(minus some changes in variable names), so I expect everything to work as
intended.

**Equicord** needs testing, as I currently lack the bandwidth to test it.
Equibop is tested, and seems to work as intended. Though I'll CC @TanvirOnGH for
further testing and possibly @Scrumplex for their review as we base on Vencord
and Vesktop.

Fixes #333175

P.S. Darwin support is omitted as I **do not** have a Darwin machine, and
therefore will not make an effort to support it. Please open a pull request and
add yourself as a maintaier if Darwin support is important to you.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See
  [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests)
    (look inside
    [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or
    [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in
    [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or
    [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are
    [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package)
    to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using
      `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all
      changes have to be committed, also see
      [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in
      `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md)
  (or backporting
  [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md)
  and
  [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md)
  Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or
        breaking
  - [ ] (Module updates) Added a release notes entry if the change is
        significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS
        module
- [x] Fits
      [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by pyrox0 — APPROVED** [review]
Been using this on my machine for a while, lgtm

**Review by ghost — APPROVED** [review]
no issues so far

File: pkgs/by-name/eq/equibop/package.nix [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  };
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs)
+      pname
+      version
+      src
+      patches
+      ;
+    hash = "sha256-2YUlkclcaZNNH12Vtp6srLgxLwXd6s2QxcoB73cIrWc=";
```
Comment (Scrumplex) [inline pkgs/by-name/eq/equibop/package.nix:43]: Please rebase and regenerate hashes.

See https://github.com/NixOS/nixpkgs/pull/349093#issuecomment-2421873637

File: pkgs/by-name/eq/equicord/package.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  src = fetchFromGitHub {
+    owner = "Equicord";
+    repo = "Equicord";
+    rev = "440b68ea82b6fd44bf5ec70b759a0207ee9f4ca7";
+    hash = "sha256-9GIw8g2HZ6/5Lb4gtDyuBqZWi5YK5Uz0lo+u+LrIZwI=";
+  };
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname src;
+    hash = "sha256-V+KY1RWxe3k69cj17IK39tM2BR0FPSgtTjC3GtUiZss=";
```
Comment (Scrumplex) [inline pkgs/by-name/eq/equicord/package.nix:23]: Same here

**Review by Scrumplex — CHANGES_REQUESTED** [review]

File: pkgs/by-name/eq/equibop/package.nix (hunk 1 above)
Comment (NotAShelf) [reply pkgs/by-name/eq/equibop/package.nix]: Done and done.

**Review by ghost — APPROVED** [review]

**Review by Scrumplex — APPROVED** [review]

**Comment by Scrumplex** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349863`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>equibop</li>
    <li>equicord</li>
  </ul>
</details>

**Pull request merged**